dalite_xblock/dalite_xblock.py

```diff
@@ -156,14 +190,74 @@ def student_view(self, context):
         :returns: XBlock HTML fragment
         :rtype: xblock.fragment.Fragment
         """
-        if not self.is_lti_ready:
-            fragment = Fragment()
-            context.update(self._get_context_for_template())
-            fragment.add_content(loader.render_django_template('/templates/xblock_data_not_filled.html', context))
-            return fragment
-        fragment = super(DaliteXBlock, self).student_view(context)
-        fragment.add_javascript(loader.load_unicode('public/js/dalite_xblock.js'))
-        fragment.initialize_js('DaliteXBlock')
+        return self._render_student_view(context, False)
+
+    @contextlib.contextmanager
+    def add_extra_custom_params(self, additional_custom_parameters):
+        """
+        Temporarily adds custom parameters to this xblock.
+
+        These parameters will be sent to the `lti` provider by any calls made during that time.
+
+        :param list additional_custom_parameters: A list of parameters in a 'key=value` format.
+               Eg: ``[u'action=launch-admin']``
+        """
+        current_params = self.custom_parameters
+        try:
+            self.custom_parameters = list(self.custom_parameters) + additional_custom_parameters
+            yield
+        finally:
+            self.custom_parameters = current_params
+
+    @XBlock.handler
+    def lti_launch_handler(self, request, suffix=u''):
+        """
+        Override superclass method.
+
+        This method interprets suffix parameter and translates it to
+        action LTI parameter, which is then passed to dalite.
+        """
+        suffix = unicode(suffix)
+        custom_params = []
+        if suffix == self.ADMIN_URL_SUFFIX:
+            custom_params = [u'action=launch-admin']
+        elif suffix == self.EDIT_QUESTION_SUFFIX:
+            custom_params = [u'action=edit-question']
+
+        with self.add_extra_custom_params(custom_params):
+            return super(DaliteXBlock, self).lti_launch_handler(request)
```

Comment:
This looks a little convoluted, but I don't think it's worth changing.  Just expressing my thoughts here, no action required.

So your proposed sequence is roughly:  XBlock form adds Suffix -> converts suffix to custom action param -> Dalite app reads custom action param -> changes authenticated redirect URL based on action param..

The only other way I can think of would be to add a couple more URL endpoints to Dalite, which lets the app respond to the `form_url_suffix` directly, e.g. update [dalite/url.py](https://github.com/open-craft/dalite-ng/blob/master/dalite/urls.py):

```diff
--- a/dalite/urls.py
+++ b/dalite/urls.py
@@ -10,11 +10,16 @@ from django.utils.translation import ugettext_lazy as _
 import views
 
 admin.site.site_header = admin.site.site_title = _('Dalite NG administration')
+from django_lti_tool_provider import views as lti_views
 
 urlpatterns = [
     url(r'', include('peerinst.urls')),
     url(r'^grappelli/', include('grappelli.urls')),
     url(r'admin_index_wrapper/', views.admin_index_wrapper, name='admin_index_wrapper'),
     url(r'^admin/', include(admin.site.urls)),
+    url(r'^lti/admin', lti_views.LTIView.as_view(),
+        {'action': 'launch-admin'}, name="lti-admin"),
+    url(r'^lti/question', lti_views.LTIView.as_view(),
+        {'action': 'edit-question'}, name="lti-edit-question"),
     url(r'^lti/', include('django_lti_tool_provider.urls')),
 ] + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
```

and using the resulting `request.kwargs.get('action')` instead of the LTI `custom_action` param.

But since this only gets rid of a small amount of code (really only the `lti_launch_handler()` overridden method), it's not really any different.